src/main/java/seedu/address/logic/commands/AddCommand.java

```diff
@@ -35,6 +41,11 @@
     public AddCommand(String name, String deadline, Set<String> tags,String freq)
             throws IllegalValueException {
         final Set<Tag> tagSet = new HashSet<>();
+        DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+        if(!deadline.matches(DATE_VALIDATION_REGEX)&&!deadline.equals("")){
+        List<Date> date = new com.joestelmach.natty.Parser().parse(deadline).get(0).getDates();
```

Comment:
What if it is an invalid date that cannot be parsed by natty? i.e. d/hi
In this case, new com.joestelmach.natty.Parser().parse(deadline) will return an empty array list which cause IndexOutOfBoundsException for get(0). 
So perhaps you can check the size first?